### ISSUE TITLE
Support custom allocator

### ISSUE THREAD

**Issue by stevefan1999-personal** [orium/rpds#67]

**Comment by elankath** [comment]
I agree that it would be good to have support for custom allocators for rpds. Maybe define allocation strategy as a trait so people can implement the trait and then use their own custom allocator: slab, slotmap, bumpalo, etc ?
